src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborWriterTests.cs

```diff
@@ -152,12 +152,45 @@ public static void WriteEncodedValue_RootValue_HappyPath(string hexEncodedValue)
             Assert.Equal(hexEncodedValue, hexResult.ToLower());
         }
 
+        [Theory]
+        [InlineData(42)]
+        [InlineData("value1")]
+        [InlineData(new object[] { new object[] { 1, 2, 3 } })]
+        public static void EncodeSpan_HappyPath(object value)
+        {
+            var writer = new CborWriter();
+            Helpers.WriteValue(writer, value);
+
+            byte[] encoding = writer.Encode();
+            byte[] target = new byte[encoding.Length];
```

Comment:
I think this would be better (standalone) written as using the `BytesWritten` property and `Encode(Span)` then later calling `Encode()` and comparing them.  That shows the relationship of the property and the method, and then circles back to to the two methods.